### DISCUSSION TITLE
IndexError: Encountered an index error.

### DISCUSSION THREAD

**Discussion by AlizadehF** [pyg-team/pytorch_geometric#8821] [Q&A]
Hello everyone

I have an error in the heterogeneous GNN model with this type:
IndexError: Encountered an index error. Please ensure that all indices in 'edge_index' point to valid indices in the interval [0, 161] (got interval [94384626, 3232300794])

I don't know where is this interval [94384626, 3232300794] from?
can you help me?

**Comment by AlizadehF** [comment]
data.validate()

ValueError: 'edge_index' of edge type ('src', 'to', 'dst') contains larger source indices than the number of nodes (162) of this node type in 'HeteroData' (found 3232300794)

**Comment by wsad1** [comment]
Thanks for sharing the output of `data.validate`. It looks like you have invalid edges in your `edge_index` , clean up the edge_index before passing the graph to the GNN.
